File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/ArrayUtil.java
package ch.rmy.android.http_shortcuts.utils;

public class ArrayUtil {

    public static int findIndex(Object[] items, Object item) {
        for (int i = 0; i < items.length; i++) {
            if (equals(items[i], item)) {
                return i;
            }
        }
        return 0;
    }

    private static boolean equals(Object a, Object b) {
        return a == null ? b == null : a.equals(b);
    }

    public static int findIndex(int[] items, int item) {
        for (int i = 0; i < items.length; i++) {
            if (items[i] == item) {
                return i;
            }
        }
        return 0;
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/ArrayUtil.kt
package ch.rmy.android.http_shortcuts.utils

object ArrayUtil {

    fun <T> findIndex(items: Array<T>, item: T) =
            items.indices.firstOrNull { items[it] == item } ?: 0

    fun findIndex(items: IntArray, item: Int) =
            items.indices.firstOrNull { items[it] == item } ?: 0

}
